chrome/browser/android/adblock/regex_resource_type_detector.cpp

```diff
@@ -0,0 +1,51 @@
+/*
+ * This file is part of Adblock Plus <https://adblockplus.org/>,
+ * Copyright (C) 2006-present eyeo GmbH
+ *
+ * Adblock Plus is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License version 3 as
+ * published by the Free Software Foundation.
+ *
+ * Adblock Plus is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with Adblock Plus.  If not, see <http://www.gnu.org/licenses/>.
+ */
+
+#include "chrome/browser/android/adblock/regex_resource_type_detector.h"
+#include <regex>
+
+static const std::regex RE_JS("\\.js$", std::regex_constants::ECMAScript | std::regex_constants::icase);
```

Comment:
We use this regex in libadblockplus-android since the very beginning:
https://github.com/adblockplus/libadblockplus-android/blob/master/adblock-android-webview/src/org/adblockplus/libadblockplus/android/webview/AdblockWebView.java#L92

But urls like ".js?smth" are not recognized as JS. Should we update regex?
If "yes" what about using of `\\.js(\?.+)*$`

Reply:
Updated regex checked to be working:
````
08-02 14:55:29.511 30826-30898/org.chromium.chrome W/chromium: [WARNING:chrome_network_delegate.cc(347)] Adblock: OnBeforeStartTransaction
    [WARNING:chrome_network_delegate.cc(356)] Adblock: isAdBlockEnabled = true, FilterEngine ptr = -1059308416
    [WARNING:chrome_network_delegate.cc(362)] Adblock: loading url https://mediator.imgsmail.ru/2/mpf-mediator.min.js
    [WARNING:chrome_network_delegate.cc(372)] Adblock: casted to AdblockPlus::FilterEnginePtr, use_count = 3
    [WARNING:chrome_network_delegate.cc(384)] Adblock: resource type of https://mediator.imgsmail.ru/2/mpf-mediator.min.js is 3 (isImage = false)
    [WARNING:chrome_network_delegate.cc(391)] Adblock: Referrer = https://m.e1.ru/
08-02 14:55:29.512 30826-30898/org.chromium.chrome W/chromium: [WARNING:chrome_network_delegate.cc(400)] Adblock: invoking IsDocumentWhitelisted(https://mediator.imgsmail.ru/2/mpf-mediator.min.js)
08-02 14:55:29.515 30826-30898/org.chromium.chrome W/chromium: [WARNING:chrome_network_delegate.cc(443)] Adblock: mapped to adblock content type 2
08-02 14:55:29.525 30826-30898/org.chromium.chrome W/chromium: [WARNING:chrome_network_delegate.cc(460)] Adblock: exiting OnBeforeStartTransaction()
08-02 14:55:29.561 30826-30898/org.chromium.chrome W/chromium: [WARNING:chrome_network_delegate.cc(347)] Adblock: OnBeforeStartTransaction
08-02 14:55:29.562 30826-30898/org.chromium.chrome W/chromium: [WARNING:chrome_network_delegate.cc(356)] Adblock: isAdBlockEnabled = true, FilterEngine ptr = -1059308416
    [WARNING:chrome_network_delegate.cc(362)] Adblock: loading url https://cstatic.weborama.fr/js/wam/customers/wamfactory_dpm.wildcard.min.js?rnd=1533203729553
    [WARNING:chrome_network_delegate.cc(372)] Adblock: casted to AdblockPlus::FilterEnginePtr, use_count = 3
    [WARNING:chrome_network_delegate.cc(384)] Adblock: resource type of https://cstatic.weborama.fr/js/wam/customers/wamfactory_dpm.wildcard.min.js?rnd=1533203729553 is 3 (isImage = false)
````

`3` is script:
https://cs.chromium.org/chromium/src/content/public/common/resource_type.h?q=ResourceType&sq=package:chromium&g=0&l=17

Reply:
or even do the group non-capturing to be consistent with others: `\\.js(?:\\?.+)*$`

Reply:
I think it's fine to use `\\.js(?:\\?.+)*$`